pythonwhat/State.py

```diff
@@ -79,10 +79,12 @@ def __init__(
         self.params = list()
 
         for k, v in args.items():
-            if k not in ["self", "args"]:
+            if k != "self":
                 self.params.append(k)
                 setattr(self, k, v)
 
+        self.messages = messages if messages else []
```

Comment:
Dangerous move in general. I'd argue it's better to check for `None` specifically. (`... if message is not None else ...`).